app/v2/controllers/V2MovementsController.scala

```diff
@@ -440,20 +444,30 @@ class V2MovementsControllerImpl @Inject() (
   ): EitherT[Future, PresentationError, BodyAndContentType] = {
     def bodyExists(bodyAndSize: BodyAndSize): EitherT[Future, PresentationError, BodyAndContentType] =
       VersionedRouting.formatAccept(acceptHeader) match {
-        case VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON if bodyAndSize.size > config.smallMessageSizeLimit =>
-          EitherT.leftT[Future, BodyAndContentType](
-            PresentationError.notAcceptableError(s"Messages larger than ${config.smallMessageSizeLimit} bytes cannot be retrieved in JSON")
-          )
-        case VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_JSON =>
-          for {
-            jsonStream <- conversionService.xmlToJson(messageSummary.messageType.get, bodyAndSize.body).asPresentation
-            bodyWithContentType = BodyAndContentType(MimeTypes.JSON, jsonStream)
-          } yield bodyWithContentType
-        case VersionedRouting.VERSION_2_ACCEPT_HEADER_VALUE_XML =>
-          EitherT.rightT[Future, PresentationError](
-            BodyAndContentType(MimeTypes.XML, bodyAndSize.body)
-          )
+        case Left(error) => EitherT.leftT(error)
+        case Right(value) =>
+          value match {
+            case VERSION_2_ACCEPT_HEADER_VALUE_JSON if bodyAndSize.size > config.smallMessageSizeLimit =>
+              EitherT.leftT[Future, BodyAndContentType](
+                PresentationError.notAcceptableError(s"Messages larger than ${config.smallMessageSizeLimit} bytes cannot be retrieved in JSON")
+              )
+            case VERSION_2_ACCEPT_HEADER_VALUE_JSON =>
+              for {
+                jsonStream <- conversionService.xmlToJson(messageSummary.messageType.get, bodyAndSize.body).asPresentation
+                bodyWithContentType = BodyAndContentType(MimeTypes.JSON, jsonStream)
+              } yield bodyWithContentType
+            case VERSION_2_ACCEPT_HEADER_VALUE_XML =>
+              EitherT.rightT[Future, PresentationError](
+                BodyAndContentType(MimeTypes.XML, bodyAndSize.body)
+              )
+            //TODO: These two cases were not caught previously should we be handling these?
```

Comment:
I don't think so it's required as if you look on entry point for this method call i.e. ```getMessageBody``` we already have ```acceptHeaderActionProvider``` that handle all case condition.  Also I think we need to use ```VERSION_2_ACCEPT_HEADER_VALUE_JSON.value```  for above case conditions?

Reply:
We may not need to handle it in terms of functionality. However, the match statement should be exhaustive, We should require it to have `.value` because the `formatAccept` now returns the models instead of a string